agent/usershell/usershell.go

```diff
@@ -50,7 +50,17 @@ func (SystemEnvInfo) User() (*user.User, error) {
 }
 
 func (SystemEnvInfo) Environ() []string {
-	return os.Environ()
+	var env []string
+	for _, e := range os.Environ() {
+		// Ignore GOTRACEBACK=none, as it disables stack traces, it can
+		// be set on the agent due to changes in capabilities.
+		// https://pkg.go.dev/runtime#hdr-Security.
+		if e == "GOTRACEBACK=none" {
```

Comment:
Is there any way for us to override this for ourselves that doesn't also override it for any user of coder? Seems like we're choosing for any user of coder that GOTRACEBACK shouldn't be set.

Reply:
On Linux, we could theoretically read `/proc/self/environ` to figure out if it was set externally or by the Go runtime. I'm not sure it's worth the lift though.

If someone sets this env in, say, a docker container resource. Yes, it will be filtered out for child-processes (which arguably may be the right thing to do anyway). But if someone sets it on the agent resource (which is the recommended way anyway), they absolutely still can.